### ISSUE TITLE
MongoDB authentication issue

### ISSUE THREAD

**Issue by uchauhan1994** [emqx/emqx#9085]
### What happened?

we have upgrade our MongoDB version 5 to 6 and we are facing issue with authentication. earlier in MongoDB version 5 authentication and access control working fine but after upgrading MongoDB version 6 then we faced authentication issue. 

### What did you expect to happen?

authentication should happens in any version of mongoDB.

### EMQX version

<details>

```console
$ ./bin/emqx_ctl broker
# paste output here
```
we are using v4.4.1
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Log files

<details>
[error] MzA3MDU4Njk1OTU0ODUzNzIzMTgyNDI0NDUwNTg0NDEyMTG@xxx.xxx.xxx.xxx:xxxxx [Hooks] Failed to execute {fun emqx_auth_mongo:check/3,[#{authquery => {authquery,<<"mqtt_user">>,[<<"password">>],sha256,[{<<"username">>,<<"%u">>}]},pool => emqx_auth_mongo,superquery => {superquery,<<"mqtt_user">>,<<"is_superuser">>,[{<<"username">>,<<"%u">>}]}}]}: {error,{bad_query,#{<<"$err">> => <<"OP_QUERY is no longer supported. The client driver may require an upgrade. For more details see https://dochub.mongodb.org/core/legacy-opcode-removal">>,<<"code">> => 5739101,<<"ok">> => 0.0}},[{mc_connection_man,process_error,2,[{file,"mc_connection_man.erl"},{line,76}]},{mc_connection_man,read_one,2,[{file,"mc_connection_man.erl"},{line,33}]},{poolboy,transaction,3,[{file,"poolboy.erl"},{line,84}]},{emqx_auth_mongo,check,3,[{file,"emqx_auth_mongo.erl"},{line,47}]},{emqx_hooks,safe_execute,2,[{file,"emqx_hooks.erl"},{line,191}]},{emqx_hooks,do_run_fold,3,[{file,"emqx_hooks.erl"},{line,170}]},{emqx_access_control,authenticate,1,[{file,"emqx_access_control.erl"},{line,80}]},{emqx_channel,auth_connect,2,[{file,"emqx_channel.erl"},{line,1302}]},{emqx_misc,pipeline,3,[{file,"emqx_misc.erl"},{line,135}]},{emqx_channel,handle_in,2,[{file,"emqx_channel.erl"},{line,303}]},{emqx_connection,with_channel,3,[{file,"emqx_connection.erl"},{line,696}]},{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,396}]},{emqx_connection,process_msg,2,[{file,"emqx_connection.erl"},{line,402}]},{emqx_connection,handle_recv,3,[{file,"emqx_connection.erl"},{line,360}]},{proc_lib,wake_up,3,[{file,"proc_lib.erl"},{line,236}]}]}
 
</details>

**Comment by lafirest** [comment]
@uchauhan1994 @zmstone 
Currently our mongo Deliver does not support newer versions of mongo

**Comment by uchauhan1994** [comment]
which version of MongoDB support EMQX v5.0.8.
any plan to support MongoDB version 6?

**Comment by xtianus79** [comment]
This is extremely important. Mongo 6 has been out now and people are going to upgrade, start or transition to 6 and effectively this is a really important issue.

**Comment by thalesmg** [comment]
Hi, sorry for the late reply.

Support for MongoDB 6 was recently added, and the auth problem was fixed [here](https://github.com/emqx/emqx/pull/10098).

This will be included in the next release (5.0.21).

**Comment by thalesmg** [comment]
Closing as the fix has been released.
